### PR TITLE
Add terraform deployment definition

### PR DESCRIPTION
**Pull request by tdooner**
Terraform is a great tool for specifying deployment configuration. As a
swiss army knife of API integrations, it can make it easy to make sure
that changes across complex deployments are made consistently, and to
introduce code review before such changes are made. Devops!

This commit adds just enough terraform configuration to create the AWS
VPC networking to get to a publicly-accessible web server. Subsequent
commits will set up RDS, additional servers, ELB's, etc. It should even
be possible to provision twilio, newrelic, and mailgun resources
declaratively with terraform, so a new deployment of clientcomm would be
just a `terraform apply` away.

More information in the deploy/README.md.

### CODE REVIEW

File: deploy/README.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# Clientcomm deploy configuration
+
+The configuration in this folder will help us deploy clientcomm in a more
```
Comment (tmaybe) [inline deploy/README.md]: Please finish this sentence :)

File: deploy/clientcomm.tf [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+resource "aws_subnet" "clientcomm_web" {
+  vpc_id = "${aws_vpc.clientcomm.id}"
+  map_public_ip_on_launch = true
+  cidr_block = "10.0.1.0/24"
+}
+
+resource "aws_security_group" "clientcomm_allow_web" {
+  name = "ClientComm Allow Web"
+  description = "Allow 80/443 to our web servers"
```
Comment (tmaybe) [inline deploy/clientcomm.tf]: 443's not allowed, but 22 is.

File: deploy/README.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+## credentials checklist
+All credentials should be exported as environment variables.
```
Comment (tmaybe) [inline deploy/README.md]: maybe recommend a method for exporting environment variables on execution, like `env $(cat .env) terraform plan` (assuming that'll work)

File: deploy/clientcomm.tf [hunk 4]
```diff
@@ -0,0 +1,86 @@
+// This is a terraform definition for an AWS configuration of clientcomm.
+// See the README.md for more details on how to use this.
+
```
Comment (tmaybe) [inline deploy/clientcomm.tf]: all the below seems pretty straightforward, but could you add brief comments to explain what each step does?

File: deploy/README.md (hunk 1 above)
Comment (tdooner) [inline deploy/README.md]: Oops, will do. 😅

File: deploy/README.md (hunk 3 above)
Comment (tdooner) [inline deploy/README.md]: Yep, that should work. That is how I do it sometimes. I'll make a note.

File: deploy/clientcomm.tf (hunk 2 above)
Comment (tdooner) [inline deploy/clientcomm.tf]: Will do.

File: deploy/clientcomm.tf (hunk 4 above)
Comment (tdooner) [inline deploy/clientcomm.tf]: Sure!

**Pull request merged**